substratevm/src/com.oracle.svm.hosted.jdk11/src/com/oracle/svm/hosted/jdk11/ClassLoaderSupportImplJDK11OrLater.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.hosted.jdk11;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.graalvm.compiler.serviceprovider.JavaVersionUtil;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import com.oracle.svm.core.ClassLoaderSupport;
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import com.oracle.svm.hosted.FeatureImpl;
+import com.oracle.svm.hosted.NativeImageClassLoaderSupport;
+import com.oracle.svm.hosted.NativeImageSystemClassLoader;
+
+import jdk.internal.module.Modules;
+
+public final class ClassLoaderSupportImplJDK11OrLater extends ClassLoaderSupport {
+
+    private final NativeImageClassLoaderSupport classLoaderSupport;
+    private final Map<String, Set<Module>> packageToModules;
+
+    ClassLoaderSupportImplJDK11OrLater(NativeImageClassLoaderSupport classLoaderSupport) {
+        this.classLoaderSupport = classLoaderSupport;
+        packageToModules = new HashMap<>();
+        buildPackageToModulesMap(classLoaderSupport);
+    }
+
+    @Override
+    protected boolean isNativeImageClassLoaderImpl(ClassLoader loader) {
+        return loader == classLoaderSupport.getClassLoader() || loader instanceof NativeImageSystemClassLoader;
+    }
+
+    @Override
+    public List<ResourceBundle> getResourceBundle(String bundleSpec, Locale locale) {
+        String[] specParts = bundleSpec.split(":", 2);
+        String moduleName;
+        String bundleName;
+        if (specParts.length > 1) {
+            moduleName = specParts[0];
+            bundleName = specParts[1];
+        } else {
+            moduleName = null;
+            bundleName = specParts[0];
+        }
+        String packageName = packageName(bundleName);
+        if (packageName == null) {
+            throw new MissingResourceException("ResourceBundle does not seem to be a fully qualified class name.", bundleName, locale.toLanguageTag());
```

Comment:
Hi @olpaw ,

With this check in place jar files including resource bundles in .properties file format at their root are not supported. Such an example is [org.eclipse.yasson](https://github.com/eclipse-ee4j/yasson/tree/master/src/main/resources).

Going through [ResourceBundle's Javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ResourceBundle.html) I was not able to find any mention/guidance on avoiding adding ResourceBundles to the root of jar files, so I am wondering if this is something we should relax on GraalVM.

Thanks

Reply:
Yes we should probably add fallback code for that to support that. Please create an issue for me.

Reply:
Hmm I forgot to mention the issue here.
FWIW the created issue was https://github.com/oracle/graal/issues/3535 and it got fixed by https://github.com/oracle/graal/pull/3540